strawberryfields/circuitspecs/gaussian_unitary.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Circuit specifications for the Gaussian simulator backend."""
+
+import numpy as np
+from strawberryfields.program_utils import Command
+from strawberryfields import ops
+from strawberryfields.parameters import par_evaluate
+from thewalrus.symplectic import expand_vector, expand, rotation, squeezing, two_mode_squeezing, interferometer, beam_splitter
+from .circuit_specs import CircuitSpecs
+
+class GaussianUnitary(CircuitSpecs):
+    """Circuit specifications for the Gaussian Unitary compiler"""
+
+    short_name = "gaussian_unitary"
+    modes = None
+    local = True
+    remote = True
+    interactive = True
+
+    primitives = {
+        # meta operations
+        "All",
+        "_New_modes",
+        "_Delete",
+        # single mode gates
+        "Dgate",
+        "Sgate",
+        "Rgate",
+        # multi mode gates
+        "BSgate",
+        "S2gate",
+        "Interferometer", # Note that interferometer is accepted as a primitive
+        "GaussianTransform", # Note that GaussianTransform is accepted as a primitive
+    }
+
+    decompositions = {
+        "GraphEmbed": {},
+        "BipartiteGraphEmbed": {},
+        "Gaussian": {},
+        "Pgate": {},
+        "CXgate": {},
+        "CZgate": {},
+        "MZgate": {},
+        "Xgate": {},
+        "Zgate": {},
+        "Fouriergate": {},
+    }
+    #pylint: disable=too-many-branches
+    def compile(self, seq, registers):
+        """Try to arrange a quantum circuit into the canonical Symplectic form.
+
+        This method checks whether the circuit can be implemented as a sequence of Gaussian operations.
+        If the answer is yes it arranges them in the canonical order with displacement at the end.
+
+
+        Args:
+            seq (Sequence[Command]): quantum circuit to modify
+            registers (Sequence[RegRefs]): quantum registers
+        Returns:
+            List[Command]: modified circuit
+        Raises:
+            CircuitError: the circuit does not correspond to a Gaussian unitary
+        """
+
+        # Check which modes are actually being used
+        used_modes = []
+        for operations in seq:
+            modes = [modes_label.ind for modes_label in operations.reg]
+            used_modes.append(modes)
+        # pylint: disable=consider-using-set-comprehension
```

Comment:
How come you don't use set comprehension here?

Reply:
I don't think it makes a difference.